tests/expose_test.go

```diff
@@ -84,34 +116,72 @@ var _ = Describe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:comp
+				Expect(cleanupJob(job.GetName(), job.GetNamespace())).To(Succeed(), "should be able to delete the batchv1.Job entity")
+			})
+
+			AfterEach(func() {
+				Expect(cleanupService(serviceName, tcpVM.GetNamespace())).To(Succeed(), "should be able to delete the k8sv1.Service entity")
+			})
+
+			table.DescribeTable("[test_id:1531][label:masquerade_binding_connectivity]Should expose a Cluster IP service on a VMI and connect to it", func(ipFamily k8sv1.IPFamily) {
 				By("Exposing the service via virtctl command")
-				virtctl := tests.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "virtualmachineinstance", "--namespace",
-					tcpVM.Namespace, tcpVM.Name, "--port", servicePort, "--name", serviceName, "--target-port", strconv.Itoa(testPort))
-				err := virtctl()
-				Expect(err).ToNot(HaveOccurred())
+				virtctl := tests.NewRepeatableVirtctlCommand(getVirtctlExposeCommand(vmiExposeArgs, ipFamily)...)
+				Expect(virtctl()).To(Succeed(), "should expose a service via `virtctl expose ...`")
 
 				By("Getting back the cluster IP given for the service")
 				svc, err := virtClient.CoreV1().Services(tcpVM.Namespace).Get(serviceName, k8smetav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
 				serviceIP := svc.Spec.ClusterIP
 
 				By("Starting a job which tries to reach the VMI via ClusterIP")
-				job := runHelloWorldJob(serviceIP, servicePort, tcpVM.Namespace)
+				job = runHelloWorldJob(serviceIP, servicePort, tcpVM.Namespace)
 
 				By("Waiting for the job to report a successful connection attempt")
 				Expect(tests.WaitForJobToSucceed(job, 420*time.Second)).To(Succeed())
-			})
+			},
+				table.Entry("over default IPv4 IP family", k8sv1.IPv4Protocol),
+				table.Entry("over IPv6 IP family", k8sv1.IPv6Protocol),
+			)
 		})
 
 		Context("Expose ClusterIP service with string target-port", func() {
 			const servicePort = "27017"
 			const serviceName = "cluster-ip-target-vmi"
-			It("[test_id:1532]Should expose a ClusterIP service and connect to the vm on port 80", func() {
-				virtctl := tests.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "virtualmachineinstance", "--namespace",
-					tcpVM.Namespace, tcpVM.Name, "--port", servicePort, "--name", serviceName, "--target-port", "http")
-				err := virtctl()
-				Expect(err).ToNot(HaveOccurred())
+
+			var vmiExposeArgs []string
+
+			BeforeEach(func() {
+				vmiExposeArgs = []string{
+					expose.COMMAND_EXPOSE,
+					"virtualmachineinstance", "--namespace", tcpVM.GetNamespace(), tcpVM.GetName(),
+					"--port", servicePort, "--name", serviceName, "--target-port", "http",
+				}
+			})
+
+			AfterEach(func() {
+				Expect(cleanupService(serviceName, tcpVM.GetNamespace())).To(Succeed(), "should be able to delete the k8sv1.Service entity")
+			})
+
+			table.DescribeTable("[test_id:1532]Should expose a ClusterIP service and connect to the vm on port 80", func(ipFamily k8sv1.IPFamily) {
```

Comment:
I would move the "[test_id:1532]" to the ipv4 entry only. We don't know if the corresponding test covers ipv6 as well. The qe guys should go over it and update the test-ids if it is.

Reply:
same regarding the missing skip.